### PR TITLE
[Bluetooth] Provide bluetooth functions

### PR DESCRIPTION
**Pull request by Wootak**
[Bluetooth] Provide bluetooth functions

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [2afe041f306011379e166012dc718c57bd340516] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2afe041f306011379e166012dc718c57bd340516] - Code Rule Check OK.

File: os/include/tinyara/bluetooth/conn.h [hunk 1]
```diff
@@ -0,0 +1,787 @@
...
+ *
+ *  @param type Connection Type
+ *  @param role Connection Role
+ *  @param id Which local identity the connection was created with
+ *  @param le LE Connection specific Info
+ *  @param br BR/EDR Connection specific Info
+ */
+struct bt_conn_info {
+	uint8_t type;
+
```
Comment (NilabhShant) [inline os/include/tinyara/bluetooth/conn.h]: Please remove this extra line.

File: os/include/tinyara/bluetooth/conn.h [hunk 2]
```diff
@@ -0,0 +1,787 @@
...
+	uint8_t role;
+
+	uint8_t id;
+
+	union {
+		struct bt_conn_le_info le;
+
```
Comment (NilabhShant) [inline os/include/tinyara/bluetooth/conn.h]: same as above.

File: os/net/bluetooth/conn_internal.h [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+	BT_CONN_CONNECT_DIR_ADV,
+	BT_CONN_CONNECT,
+	BT_CONN_CONNECTED,
+	BT_CONN_DISCONNECT,
+} bt_conn_state_t;
+
+/* bt_conn flags: the flags defined here represent connection parameters */
+enum {
+	BT_CONN_AUTO_CONNECT,
+	BT_CONN_BR_LEGACY_SECURE,	/* 16 digits legacy PIN tracker */
```
Comment (NilabhShant) [inline os/net/bluetooth/conn_internal.h]: Please aline comments.

File: os/include/tinyara/bluetooth/bluetooth.h [hunk 4]
```diff
@@ -141,4 +141,462 @@ struct iob_s;					/* Forward reference */
...
+	uint8_t type;
+	uint8_t data_len;
+	const uint8_t *data;
+};
+
+/** OOB data that is specific for LE SC pairing method. */
+struct bt_le_oob_sc_data {
+	/** Random Number. */
+	uint8_t r[16];
+
```
Comment (NilabhShant) [inline os/include/tinyara/bluetooth/bluetooth.h]: Please remove this extra line.

File: os/include/tinyara/bluetooth/bluetooth.h [hunk 5]
```diff
@@ -141,4 +141,462 @@ struct iob_s;					/* Forward reference */
...
+};
+
+/** @brief BR/EDR discovery result structure */
+struct bt_br_discovery_result {
+	/** private */
+	uint8_t _priv[4];
+
+	/** Remote device address */
+	bt_addr_t addr;
+
```
Comment (NilabhShant) [inline os/include/tinyara/bluetooth/bluetooth.h]: same as above.

**Comment by seinfra** [comment]
Target : [7ac261d493950c247323b26412186fb3238ff419] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7ac261d493950c247323b26412186fb3238ff419] - Code Rule Check OK.

File: os/net/bluetooth/bluetooth.c [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include <string.h>
+
```
Comment (drashti304) [inline os/net/bluetooth/bluetooth.c]: Please remove extra line

**Comment by seinfra** [comment]
Target : [4a072d4fb3a09933fdbeb03dc2134158c8ecdd8f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4a072d4fb3a09933fdbeb03dc2134158c8ecdd8f] - Code Rule Check (C++) OK.

**Pull request merged**